server_test.go

```diff
@@ -134,3 +134,25 @@ func TestServer_ServeHTTP_health(t *testing.T) {
 	assert.Equal(t, 200, w.Code)
 	assert.Equal(t, "OK\n", w.Body.String())
 }
+
+func BenchmarkServe(b *testing.B) {
+	p := new(pub)
+
+	s := Server{
+		Log:       log.New(ioutil.Discard, "", log.LstdFlags),
+		Topic:     "builds",
+		Publisher: p,
+	}
+
+	r, err := http.NewRequest("POST", "/build", bytes.NewBufferString(`{ "foo": "bar" }`))
+	if err != nil {
+		b.Error(err)
+	}
+	r.Header.Set("Content-Type", "application/json")
+
+	w := httptest.NewRecorder()
+
+	for i := 0; i < b.N; i++ {
+		s.ServeHTTP(w, r)
```

Comment:
The request body is gonna be consumed by the first call, I believe you'll have to reset it if you want it to be parsed again.